File: src_simple/python/pi.py
#!/usr/bin/env python
# -*- coding=utf-8 -*-

""" pi  -  14.03.2014

    Calcul du nombre pi par Monte Carlo.
    Principe : La probabilité qu'un point de coordonnées (x,y) 
    avec x et y dans l'intervalle [0;1] soit dans le quart de 
    cercle de centre (0,0) et de rayon 1 est égal au rapport des 
    aires du quart de cercle de rayon 1 et du carré de largeur 1 
    soit pi/4. """

import math
from random import random

def prog_pi(ntirage = 100):
    """ calcul du nombre pi """

    # initialisation
    n = 0

    # calcul monte carlo
    for i in range(ntirage):
        x = random()
        y = random()

        if x**2 + y**2 < 1.0:
            n += 1

    # affichage des résultats
    piCalcule = 4.0 * float(n) / float(ntirage)
    print("pi         = {0}".format(piCalcule))
    print("% d'erreur = {0}".format((math.pi - piCalcule) / math.pi * 100))

if __name__ == "__main__":
    # lecture du nombre de tirage
    ntirage = int(raw_input("entrer le nombre de tirage : "))
    print("ntirage = {0}".format(ntirage))

    prog_pi(ntirage)

File: src_simple/f90/pi.f90
program prog_pi

  implicit none

  integer                     :: i, n, ntirage
  double precision            :: x, y
  double precision, parameter :: pi = 3.141592653589793d0

  ! lecture du nombre de tirage
  write(*, *) "Nombre de tirage : "
  read(*, *) ntirage
  write(*, *) "ntirage = ", ntirage

  ! initialisation
  n = 0

  ! calcul du nombre pi
  do i = 1, ntirage, 1
      call random_number(x)
      call random_number(y)

      if (x**2 + y**2 < 1.d0) then
          n = n + 1
      end if
  end do

  write(*, *) "pi = ", 4.0 * dble(n) / dble(ntirage)
  write(*, *) "% d'erreur = ", (pi - 4.d0 * dble(n) / dble(ntirage)) / pi * 100.d0

end program prog_pi

